src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultEditorTemplates.cs

```diff
@@ -288,18 +281,27 @@ public static IHtmlContent ObjectTemplate(IHtmlHelper htmlHelper)
                     readOnly: false,
                     additionalViewData: null);
 
-                content.Append(templateBuilder.Build());
-
+                var templateBuilderResult = templateBuilder.Build();
                 if (!propertyMetadata.HideSurroundingHtml)
```

Comment:
The new structure may be _too_ similar to the previous code. Suggest seeing if the loop is more clear with a single `propertyMetadata.HideSurroundingHtml` check, similar to my recent `DefaultHtmlGenerator` comment. Could do the same thing in `DefaultDisplayTemplates` but there's less inside the conditional blocks there.